File: problems/algorithms/0009/C/solution_no_array.c
bool isPalindrome(int x)
{
	if (x < 0)
		return false;
	if (x < 10)
		return true;
	long reversed = 0;
	int units = 0;
	int temporary = x;
	while (temporary != 0)
	{
		units = temporary % 10;
		reversed = reversed * 10 + units;
		temporary /= 10;
	}
	return x == reversed;
}

File: problems/algorithms/0009/Rust/solution_no_array.rs
impl Solution {
	pub fn is_palindrome(x: i32) -> bool {
		if x < 0 {
			return false;
		}
		if x < 10 {
			return true;
		}
		let mut reversed = 0;
		let mut temporary = x;
		while temporary != 0 {
			let units = temporary % 10;
			reversed = reversed * 10 + units;
			temporary /= 10;
		}
		x == reversed
	}
}